### PR TITLE
Pass extra authentication params in the OIDC DevUI code flow redirect URL

### LINKED ISSUE #38489: OIDC authentication.extra-params not added to dev-services auth request
### Describe the bug

When using the Auth0 provider, i would like to get an access_token in a JWT format. To get this, auth0 says i need to set an audience query parameter when requesting the token via /authorize.

I have configured the quarkus property:
`quarkus.oidc.authentication.extra-params.audience=my-audience`

However "audience" is not added to the authentication query param:

```
https://<server-url>/authorize?client_id=<client_id>&redirect_uri=http://localhost:8080/q/dev-ui/io.quarkus.quarkus-oidc/auth0-provider&scope=openid profile&response_type=code&response_mode=query&prompt=login&nonce=yN1mogD&state=hyFUmSF
```

image

### Expected behavior

I expected anything added to quarkus.oidc.authentication.extra-params would be passed to "/authorize"

### Actual behavior

The audience param is not added to the "/authorize" route.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Linux danprueitt-fedora 6.6.13-200.fc39.x86_64 #1 SMP PREEMPT_DYNAMIC Sat Jan 20 18:03:28 UTC 2024 x86_64 GNU/Linux

### Output of `java -version`

openjdk 21.0.1 2023-10-17 LTS OpenJDK Runtime Environment Temurin-21.0.1+12 (build 21.0.1+12-LTS) OpenJDK 64-Bit Server VM Temurin-21.0.1+12 (build 21.0.1+12-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.6.8

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #38489.

For example, without this fix, OIDC Dev UI will continue showing the Auth0 access tokens in the encrypted form and such tokens won't work with the demo, while it works correctly if the demo endpoint is accessed directly from the browser in which case the extra parameters which enable Auth0 to present access tokens in the JWT format are correctly added by `quarkus-oidc`

### CODE REVIEW

**Comment by michalvavrik** [comment]
I checked code carefully, looks good, but I won't have opportunity to check it out and test it today.

**Comment by sberyozkin** [comment]
@michalvavrik Sure, take your time please, I tested it using the auth0 tutotial, with and without these extra properties set. Initially I retrieved the Map as a non-optional property to test the actual issue solution, but then at the last moment I tested without any extra params and it failed, so I made it an Optional property retrieval and it worked

**Comment by sberyozkin** [comment]
Hi @michalvavrik Can you please have a quick look next week on this one ? Please don't spend time this weekend. 

Auth0 will give binary access tokens bt default and give them in JWT format if the extra properties such as `audience` configured with the demo are passed as part of the code flow redirect.
This is described here: https://quarkus.io/guides/security-oidc-auth0-tutorial#jwt-access-tokens and it already works correctly outside of Dev UI, this issue is only about making sure that after the extra parameters have been configured, Access token is shown in JWT format and can be used to test the service. 
If you think the code looks fine then may be it is good to go if the time will be limited next week ? The image I added to this PR can only be produced with this fix

**Comment by sberyozkin** [comment]
Thanks @michalvavrik See how it goes on Mon, next week will be grand

**Review by michalvavrik — APPROVED** [review]
I gave it a try + rechecked the code, lgtm, thanks.

File: docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc [hunk 1]
```diff
@@ -804,6 +804,12 @@ public class GreetingResource {
 
 Open a browser, access http://localhost:8080/hello and get your name displayed in the browser.
 
+Lets go to http://localhost:8080/q/dev, select the `OpenId Connect` card, login to Auth0, and check the Access token content:
```
Comment (gsmet) [inline docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc]:
```suggestion
Go to http://localhost:8080/q/dev-ui, select the `OpenId Connect` card, login to Auth0, and check the Access token content:
```

**Review by gsmet — APPROVED** [review]

**Comment by sberyozkin** [comment]
Thanks @michalvavrik @gsmet

**Pull request merged**